lib/parsers/nef_parser.rb

```diff
@@ -0,0 +1,69 @@
+class FormatParser::NEFParser
+  include FormatParser::IOUtils
+  include FormatParser::EXIFParser
+
+  MAGIC_LE = [0x49, 0x49, 0x2A, 0x0].pack('C4')
+  MAGIC_BE = [0x4D, 0x4D, 0x0, 0x2A].pack('C4')
+  HEADER_BYTES = [MAGIC_LE, MAGIC_BE]
+  NEF_MIME_TYPE = 'image/x-nikon-nef'
+
+  SUBFILE_TYPE_FULL_RESOLUTION = 0
+  SUBFILE_TYPE_REDUCED_RESOLUTION = 1
+
+  def likely_match?(filename)
+    filename =~ /\.nef$/i
+  end
+
+  def call(io)
+    io = FormatParser::IOConstraint.new(io)
+
+    return unless HEADER_BYTES.include?(safe_read(io, 4))
+
+    # Because of how NEF files organize their IFDs and subIFDs, we need to dive into the subIFDs
+    # to get the actual image dimensions instead of the preview's
+    should_parse_sub_ifds = true
```

Comment:
I'd say this is a constant because it's always gonna be true for the nef_parser. So I'd put it amongst the other constants on top. What do you think? 🤔